src/psyclone/parse/kernel.py

```diff
@@ -367,52 +363,47 @@ def get_stencil(metadata, valid_types):
 
     if not isinstance(metadata, expr.FunctionVar):
         raise ParseError(
-            "Expecting format stencil(<type>[,<extent>]) but found the "
-            "literal {0}".format(metadata))
+            f"Expecting format stencil(<type>[,<extent>]) but found the "
+            f"literal {metadata}")
     if metadata.name.lower() != "stencil" or not metadata.args:
         raise ParseError(
-            "Expecting format stencil(<type>[,<extent>]) but found {0}".
-            format(metadata))
+            f"Expecting format stencil(<type>[,<extent>]) but found {metadata}"
+            )
     if len(metadata.args) > 2:
         raise ParseError(
-            "Expecting format stencil(<type>[,<extent>]) so there must "
-            "be at most two arguments inside the brackets {0}".
-            format(metadata))
+            f"Expecting format stencil(<type>[,<extent>]) so there must "
+            f"be at most two arguments inside the brackets {metadata}")
     if not isinstance(metadata.args[0], expr.FunctionVar):
-        if isinstance(metadata.args[0], six.string_types):
+        if isinstance(metadata.args[0], str):
             raise ParseError(
-                "Expecting format stencil(<type>[,<extent>]). However, "
-                "the specified <type> '{0}' is a literal and therefore is "
-                "not one of the valid types '{1}'".
-                format(metadata.args[0], valid_types))
+                f"Expecting format stencil(<type>[,<extent>]). However, "
+                f"the specified <type> '{metadata.args[0]}' is a literal and "
+                f"therefore is not one of the valid types '{valid_types}'")
         raise ParseError(
-            "Internal error, expecting either FunctionVar or "
-            "str from the expression analyser but found {0}".
-            format(type(metadata.args[0])))
+            f"Internal error, expecting either FunctionVar or str from the "
+            f"expression analyser but found {type(metadata.args[0])}")
     if metadata.args[0].args:
         raise ParseError(
             "Expected format stencil(<type>[,<extent>]). However, the "
             "specified <type> '{0}' includes brackets")
```

Comment:
The argument the f-string is missing, maybe it is meant to be `metadata.args[0]`?

Reply:
This is going to be deleted soon, so we can ignore it.